docs/quick_start/circuits/index.md

```diff
@@ -96,19 +74,34 @@ pprint(ast)
 
 ![QFT QASM2](qft-qasm2.png)
 
+
 ## Understanding the compilation process
 
 The compilation process is divided into several stages:
 
-1. **Lowering**: the decorator `qasm2.extended` takes Python Abstract Syntax Tree (AST) and lowers it into Kirin IR in the Static Single Assignment (SSA) form.
+1. **Lowering**: the decorator `qasm2.extended` takes the Python Abstract Syntax Tree (AST) and lowers it into Kirin IR in the Static Single Assignment (SSA) form.
 2. **Interpretation**: when invoking the PyQrack backend, the IR code is interpreted via Kirin's IR interpreter (missing link) with the PyQrack runtime backend.
 3. **Target code generation**: when emitting QASM2 code:
    1. The IR code gets aggressively inlined and all constant expressions are evaluated.
    2. All loops and control flow are unrolled.
    3. All compatible Python expressions (e.g `sin`, arithmetics) are translated into QASM2 expression.
    4. The QASM2 code is emitted as QASM2 AST for pretty printing.
 
-In fact, the decorator `qasm2.extended` is a group of smaller dialects:
+### Progressive compilation
+
+As well as writing circuit executions, you can also progressively transform and compile that circuit. For example, you may want to lower arbitrary single qubit unitaries into hardware-specific unitaries, as is done in [this example](quick_start/circuits/compiler_passes/native_gate_rewrite.md). For more details on the kinds of circuit-level compiler passes and how to write your own, see [here](quick_start/circuits/compiler_passes/index.md)
+
+## Dialect groups
+
+Bloqade provides a set of [dialects (missing link)]() for QASM2 and our custom extensions to model parallel gates in neutral atom architectures. The basic QASM2 functionality can be enabled via
```

Comment:
Need some kind of canonical definition for dialect we can link to, don't know if it's floating around on some ACM paper or if Wikipedia has the definition, I can't seem to find it with a cursory google search.

Reply:
just link to Kirin maybe?